modules/plottinggl/processors/scatterplotmatrixprocessor.cpp

```diff
@@ -341,14 +356,34 @@ void ScatterPlotMatrixProcessor::createLabels() {
         textRenderer_.setFontSize(fontSize_.get());
 
         auto &dataFrame = *dataFrame_.getData();
-        for (auto x = dataFrame.begin() + 1; x != dataFrame.end(); ++x) {
+        for (auto x = dataFrame.begin(); x != dataFrame.end(); ++x) {
+            if (!isIncluded(*x)) continue;
             auto tex = util::createTextTexture(textRenderer_, (*x)->getHeader(), fontSize_.get(),
                                                fontColor_.get());
             labelsTextures_.push_back(tex);
         }
     }
 }
 
+bool ScatterPlotMatrixProcessor::isIncluded(std::shared_ptr<Column> col) {
```

Comment:
I don't like that `isIncluded()` has side effects, i.e. creates a property if it doesn't exist yet.
But what happens if I connect a different DataFrame to the inport which has a column with the same name as the old DataFrame, which you had disabled. Then this particular column would not be shown, correct? Is this what we want?